.github/workflows/build-and-check.yml

```diff
@@ -67,10 +67,10 @@ jobs:
     needs: build
     steps:
       - uses: actions/checkout@v2
-      - name: set up JDK 1.8
+      - name: set up JDK 11
         uses: actions/setup-java@v2
         with:
-          java-version: 8
+          java-version: 11
           distribution: 'zulu'
```

Comment:
We'd like to maintain backwards compatibility with JDK 8, so building with 8 is what we want still.

However, if you'd like to add support for matrix testing with 8, 11, and 17, that would be cool. If you do do that, please make sure that publishing still happens with JDK 8

Reply:
But the min JDK required by AGP is 11 now.

```
* What went wrong:
An exception occurred applying plugin request [id: 'com.android.application', artifact: 'com.android.tools.build:gradle:7.0.4']
> Failed to apply plugin 'com.android.internal.application'.
   > Android Gradle plugin requires Java 11 to run. You are currently using Java 1.8.
     You can try some of the following options:
       - changing the IDE settings.
       - changing the JAVA_HOME environment variable.
       - changing `org.gradle.java.home` in `gradle.properties`.
```

Reply:
Can you reconfigure the build so that the android tests only execute when executed with the required java version? I'd still like to test the plugin against jdk 1.8 for non-android purposes. Also, does this change mean that we'll drop support for older versions of the Android plugin or are these changes backwards compatible?

@Tapchicoma you understand this much better than I do, can you weigh in here?

Reply:
😅 No idea.

Reply:
I've add `check_android_bashisms` & `check_android_samples` jobs to check android modules, CC @JLLeitschuh